In [5]:
import numpy as np
import pandas as pd
import scipy.stats as stats
from mll_calc.mll_pred import format_XY, ratios, mll_testset

In [6]:
unc = 0.05
lbls = ['ReactorType', 'CoolingTime', 'Enrichment', 'Burnup', 'OrigenReactor']
tamu_list = ['cs137/cs133', 'cs134/cs137', 'cs135/cs137', 'ba136/ba138',
             'sm150/sm149', 'sm152/sm149', 'eu154/eu153', 'pu240/pu239',
             'pu241/pu239', 'pu242/pu239']

In [7]:
train_db = '~/sims_n_results/sfcompo_compare_aug2020/ohi2_not-scaled_15nuc.pkl'
test_db = '~/sfcompo/format_clean/sfcompo_formatted.pkl'
XY = format_XY(train_db)
test = pd.read_pickle(test_db)

ratio_list = tamu_list
XY = ratios(XY, ratio_list, lbls)
test = ratios(test, ratio_list, lbls)

## Test sample from SFCOMPO that was simulated in ORIGEN

In [8]:
test = test.loc[test.index == 'OHI-2|17G|C5|89G03']
test

cs137/cs133  cs134/cs137  cs135/cs137  ba136/ba138  \
OHI-2|17G|C5|89G03          0.0     0.022164          0.0          0.0   

                    sm150/sm149  sm152/sm149  eu154/eu153  pu240/pu239  \
OHI-2|17G|C5|89G03          0.0          0.0          0.0     0.473684   

                    pu241/pu239  pu242/pu239 ReactorType  CoolingTime  \
OHI-2|17G|C5|89G03     0.281307     0.137568         pwr          0.0   

                    Enrichment   Burnup OrigenReactor  
OHI-2|17G|C5|89G03      1.6874  28717.0     Ohi-2_PWR

## 12 Sims from ORIGEN 

#### Only variations are: 
1. 12 cooling times, since that is unknown in SFCOMPO (ranging 1 min to 12 years)

#### OHI-2|17G|C5|89G0 Simulation Parameters:
* PWR: Ohi-2_PWR, so origen equivalent is W(17x17)
* Enrichment: 1.6874
* Burnup: 28717.0 MW*d/tHM
* Cooling time: 1 min --> 12 yrs
* U weight: 87 MT
* Thermal power: 3423 MW for cycles 5 & 6
* Avg power for ORIGEN is in MW/MTHM, so did 39 for sim

In [9]:
XY

cs137/cs133  cs134/cs137  cs135/cs137  ba136/ba138  sm150/sm149  \
0      0.913474     0.000981     1.034931     0.001093     1.171979   
1      0.913474     0.000981     1.034931     0.001093     1.171979   
2      0.912836     0.000981     1.035569     0.001093     1.170203   
3      0.912199     0.000980     1.035951     0.001094     1.168786   
4      0.906474     0.000975     1.036730     0.001102     1.161751   
5      0.900932     0.000956     1.038186     0.001116     1.160529   
6      0.896698     0.000901     1.042848     0.001122     1.160529   
7      0.881859     0.000718     1.060397     0.001123     1.160529   
8      0.861728     0.000525     1.085169     0.001123     1.160529   
9      0.822961     0.000281     1.136287     0.001123     1.160529   
10     0.750489     0.000080     1.246015     0.001123     1.160529   
11     0.684344     0.000023     1.366448     0.001123     1.160529   

    sm152/sm149  eu154/eu153  pu240/pu239  pu241/pu239   pu242/pu239  \
0      0.631299     0.006143     0.008105     0.000120  3.415793e-07   
1      0.631299     0.006143     0.008105     0.000120  3.415793e-07   
2      0.630343     0.006140     0.008099     0.000120  3.413556e-07   
3      0.629579     0.006137     0.008094     0.000120  3.411322e-07   
4      0.625790     0.006112     0.008070     0.000119  3.401305e-07   
5      0.625132     0.006081     0.008065     0.000119  3.399087e-07   
6      0.625132     0.005988     0.008065     0.000118  3.399087e-07   
7      0.625132     0.005649     0.008066     0.000114  3.399641e-07   
8      0.625132     0.005214     0.008065     0.000108  3.399641e-07   
9      0.625132     0.004435     0.008063     0.000098  3.399641e-07   
10     0.625132     0.003214     0.008061     0.000081  3.400196e-07   
11     0.625132     0.002328     0.008058     0.000067  3.400196e-07   

   ReactorType  CoolingTime  Enrichment   Burnup OrigenReactor  
0          pwr     0.000000        1.69  28717.0        w17x17  
1          pwr     0.000694        1.69  28717.0        w17x17  
2          pwr     0.500000        1.69  28717.0        w17x17  
3          pwr     1.000000        1.69  28717.0        w17x17  
4          pwr     7.000000        1.69  28717.0        w17x17  
5          pwr    30.000000        1.69  28717.0        w17x17  
6          pwr   100.000000        1.69  28717.0        w17x17  
7          pwr   365.250000        1.69  28717.0        w17x17  
8          pwr   730.500000        1.69  28717.0        w17x17  
9          pwr  1461.000000        1.69  28717.0        w17x17  
10         pwr  2922.000000        1.69  28717.0        w17x17  
11         pwr  4383.000000        1.69  28717.0        w17x17

## First, try to find best prediction via MLL

In [10]:
pred_df = mll_testset(XY, test, True, unc, lbls)

In [11]:
pd.set_option("display.max_rows", None, "display.max_columns", None)
pred_df

sim_idx ReactorType  CoolingTime  Enrichment   Burnup  \
0  OHI-2|17G|C5|89G03         pwr          0.0      1.6874  28717.0   

  OrigenReactor  pred_idx pred_ReactorType  pred_CoolingTime  pred_Enrichment  \
0     Ohi-2_PWR         0              pwr               0.0             1.69   

   pred_Burnup pred_OrigenReactor      MaxLogLL      MaxLLUnc   2ndMaxLogLL  \
0      28717.0             w17x17 -3.244104e+13  3.244002e+12 -3.244104e+13   

    2ndMaxLLUnc  CDF_LogLL_0.9998  CDF_LLUnc_0.9998  CDF_LogLL_0.9988  \
0  3.244002e+12     -3.244104e+13      3.275967e+12     -3.244104e+13   

   CDF_LLUnc_0.9988  CDF_LogLL_0.95  CDF_LLUnc_0.95  CDF_LogLL_0.9  \
0      3.275967e+12   -3.244104e+13    3.275967e+12  -3.244529e+13   

   CDF_LLUnc_0.9  CDF_LogLL_0.5  CDF_LLUnc_0.5  CDF_LogLL_0.1  CDF_LLUnc_0.1  \
0   3.275861e+12  -3.274132e+13   3.273831e+12  -3.275970e+13   3.244427e+12   

   CDF_LogLL_0.01  CDF_LLUnc_0.01  
0   -3.276073e+13    3.244002e+12

## Second, dig in and look at ratio diffs

In [12]:
def logpdf_calc(sim_row, test_sample, unc):
    y_sim = sim_row[test_sample>0].values.tolist()
    std = sim_row.multiply(unc)[test_sample>0].values.tolist()
    y_mes = test_sample[test_sample>0].values.tolist()
    logpdf_list = stats.norm.logpdf(y_sim, loc=y_mes, scale=std)
    return logpdf_list

In [14]:
test_sample = test.drop(columns=lbls).squeeze()
test_answer = test[lbls].squeeze()
sim_row = XY.loc[0].drop(lbls).squeeze()
logpdf = logpdf_calc(sim_row, test_sample, unc)
logpdf = pd.Series(logpdf, index=sim_row[test_sample>0].index, name='LogPDF')

In [16]:
df = pd.DataFrame([test_sample, sim_row])
df.loc['Abs Diff'] = np.abs(df.loc[0] - df.loc['OHI-2|17G|C5|89G03'])
df.loc['% Diff'] = df.loc['Abs Diff'] * 100 / df.loc['OHI-2|17G|C5|89G03']

In [17]:
df.loc['LogPDF'] = logpdf

In [18]:
df

cs137/cs133   cs134/cs137  cs135/cs137  ba136/ba138  \
OHI-2|17G|C5|89G03     0.000000      0.022164     0.000000     0.000000   
0                      0.913474      0.000981     1.034931     0.001093   
Abs Diff               0.913474      0.021183     1.034931     0.001093   
% Diff                      inf     95.573965          inf          inf   
LogPDF                      NaN -93247.592045          NaN          NaN   

                    sm150/sm149  sm152/sm149  eu154/eu153    pu240/pu239  \
OHI-2|17G|C5|89G03     0.000000     0.000000     0.000000       0.473684   
0                      1.171979     0.631299     0.006143       0.008105   
Abs Diff               1.171979     0.631299     0.006143       0.465580   
% Diff                      inf          inf          inf      98.289045   
LogPDF                      NaN          NaN          NaN -660022.500950   

                     pu241/pu239   pu242/pu239  
OHI-2|17G|C5|89G03  2.813067e-01  1.375681e-01  
0                   1.199214e-04  3.415793e-07  
Abs Diff            2.811868e-01  1.375677e-01  
% Diff              9.995737e+01  9.999975e+01  
LogPDF             -1.099580e+09 -3.243994e+13